pymodm/base/models.py

```diff
@@ -295,11 +299,13 @@ def to_son(self):
             for field in self._mongometa.get_fields():
                 if field.is_undefined(self):
                     continue
-                raw_value = self._data.get(field.attname)
-                if field.is_blank(raw_value):
-                    son[field.mongo_name] = raw_value
+                value = self._data.get_python_value(
```

Comment:
this can just be `get_mongo_value`

Reply:
So I tried to change this and realized why I had used such a convoluted conversion scheme to begin with. Consider the situation in `test.test_fields.FieldsTestCase.test_save_blank_value`:
```
    def test_save_blank_value(self):
        inst = Simple('a string with more than zero characters').save()
        inst.name = ''
        inst.save()
        self.assertEqual('', DB.simple.find_one()['name'])

        inst.name = None
        inst.save()
        self.assertIsNone(DB.simple.find_one()['name'])
```

When we do `inst.name = None`, we end up storing `None` in `inst._data._python_data['name']`. When we try to convert this model to SON during the call to `save()`, we end up running `None` through the `to_mongo` method of `CharField` which gives us `'None'` (a string) instead of the `NoneType` value `None`. Clearly, that behavior is not expected or correct. The reason we run into it is because the `LazyDecoder` does not have enough information about the fields to figure out on its own what qualifies as a 'blank' value and what does not. 

For now we should continue using the convoluted logic and revisit this at a later point when we overhaul `LazyDecoder` to make it 'smarter'.